lib/core/src/display/shape/text/cursor.rs

```diff
@@ -230,62 +238,44 @@ impl<'a,'b> CursorNavigation<'a,'b> {
 // === Cursors ===
 // ===============
 
-/// The number of vertices of single cursor.
-const CURSOR_BASE_LAYOUT_SIZE : usize = 2;
-
-lazy_static! {
-    /// The base vertices position of single cursor. This position is then translated to the
-    /// actual cursor position.
-    pub static ref CURSOR_VERTICES_BASE_LAYOUT : [Point2<f32>;CURSOR_BASE_LAYOUT_SIZE] =
-        [ Point2::new(0.0, -0.2)
-        , Point2::new(0.0,  0.8)
-        ];
-}
-
 /// Structure handling many cursors.
 ///
 /// Usually there is only one cursor, but we have possibility of having many cursors in one text
-/// component enabling editing in multiple lines/places at once. This structure also owns
-/// a WebGL buffer with vertex positions of all cursors.
+/// component enabling editing in multiple lines/places at once.
 #[derive(Debug)]
 pub struct Cursors {
+    /// All cursors' positions.
     pub cursors : Vec<Cursor>,
-    pub dirty   : bool,
-    pub buffer  : Option<WebGlBuffer>,
 }
 
-impl Cursors {
-
-    /// Create empty `Cursors` structure.
-    pub fn new(gl_context:&Context) -> Self {
+impl Default for Cursors {
+    fn default() -> Self {
         Cursors {
-            cursors : Vec::new(),
-            dirty   : false,
-            buffer  : gl_context.create_buffer()
+            cursors : vec![Cursor::new(TextLocation::at_document_begin())],
         }
     }
+}
 
-    /// Update the cursors' buffer data.
-    pub fn update_buffer_data
-    (&mut self, gl_context:&Context, content:&mut TextComponentContent, fonts:&mut Fonts) {
-        let cursors          = self.cursors.iter();
-        let cursors_vertices = cursors.map(|cursor| Self::cursor_vertices(cursor,content,fonts));
-        let buffer_data      = cursors_vertices.flatten().collect_vec();
-        set_buffer_data(gl_context,self.buffer.as_ref().unwrap(),buffer_data.as_slice());
-        self.dirty = false;
-    }
-
+impl Cursors {
     /// Removes all current cursors and replace them with single cursor without any selection.
     pub fn set_cursor(&mut self, position: TextLocation) {
         self.cursors = vec![Cursor::new(position)];
-        self.dirty   = true;
+    }
+
+    /// Remove all cursors except the active one.
+    pub fn remove_additional_cursors(&mut self) {
+        self.cursors.drain(0..self.cursors.len()-1);
+    }
+
+    /// Return the active (last added) cursor.
+    pub fn active_cursor(&self) -> &Cursor {
```

Comment:
If I understand correctly, all cursors are "active", not only the last one, right? If so, the naming is off.

Reply:
That depends. E.g when you have multiline cursor in Intellij, you edit using all the cursors, but only one is active (this one which highlights the line). Only the active cursor scrolls text on moving, and only this cursor's selection is extended on shift+alt+mouse click

Reply:
Added docs for explaining that